vignettes/epidist.Rmd

```diff
@@ -1,201 +1,250 @@
+ggplot(obs_cens, aes(x = delay, y = delay_daily)) +
+  geom_point(col = "#E69F00") +
+  geom_abline(slope = 1, intercept = 0, linetype = "dashed", col = "grey") +
+  theme_minimal() +
+  coord_fixed() +
+  labs(x = "Exact delay time (days)", y = "Censored delay time (days)")
 ```
 
-Plot the true continuous delay distribution and the empirical observed distribution for each observation window.
+Next, during an outbreak we will usually be estimating delays in real time.
+The result is that only those cases with a secondary event occurring before some time will be observed.
+This is called (right) truncation, and biases the observation process towards shorter delays:
+<!-- I think possible to do a better job with showing what right truncation is with a figure here! -->
 
-```{r empirical-dist}
-combined_obs <- combine_obs(truncated_obs, obs)
+```{r}
+obs_time <- 25
+# filter_obs_by_obs_time() should be renamed to refer to stime
+obs_cens_trunc <- filter_obs_by_obs_time(obs_cens, obs_time = obs_time)
+```
 
-plot_empirical_delay(
-  combined_obs, meanlog = secondary_dist$meanlog[[1]],
-  sdlog = secondary_dist$sdlog[[1]]
-)
+Finally, in reality, it's not possible to observe every case.
+We suppose that a sample of individuals of size `sample_size` are observed:
+
+```{r}
+sample_size <- 200
 ```
 
-Plot the mean difference between continuous and discrete event time:
+This sample size corresponds to `r 100 * round(sample_size / nrow(obs_cens_trunc), 3)`% of the data.
 
-```{r censor_delay}
-censor_delay <- calculate_censor_delay(truncated_obs)
-plot_censor_delay(censor_delay)
+```{r}
+obs_cens_trunc_samp <-
+  obs_cens_trunc[sample(seq_len(.N), sample_size, replace = FALSE)]
 ```
 
-### Model
+With our censored, truncated, and sampled data, we are now ready to try to recover the underlying delay distribution using `epidist`.
 
-Adjust for right truncation and date censoring using a latent variable approach.
+# Fit the model {#fit}
 
-```{r}
-latent_truncation_censoring_fit <- latent_truncation_censoring_adjusted_delay(
-  data = truncated_obs, cores = 4, refresh = 0
-)
+If we had access to the data `obs`, then it would be simple to estimate the delay distribution (Figure \@ref(fig:obs-est)).
+
+(ref:obs-est) The histogram of delays matches closely the underlying distribution. Edit: at the moment something is wrong here.
+
+```{r obs-est, fig.cap="(ref:obs-est)"}
+ggplot() +
+  geom_histogram(data = obs, aes(x = delay, y = stat(count / sum(count)))) +
+  geom_function(
+    data = data.frame(x = c(0, 30)),
+    aes(x = x),
+    fun = dlnorm,
+    meanlog = secondary_dist[["meanlog"]],
+    sdlog = secondary_dist[["sdlog"]],
+  ) +
+  labs(x = "Delay between primary and secondary event (days)", y = "Probability density") +
+  theme_minimal()
 ```
 
-### Summarise model posterior and compare to known truth
+However, with only access to `obs_cens_trunc_samp`, we must use a statistical model to avoid severely biased estimates!
```

Comment:
Prove it to me with a plot

Reply:
Agree and done.

Reply:
Still IDK what's going on with the true density..

![image](https://github.com/epinowcast/epidist/assets/29818116/276db2f2-7d4a-48ed-824f-231833c08fe1)

Reply:
Hmm so the version we have here: https://github.com/parksw3/epidist-paper

definitely works as intended (we should talk a bit about why we are plotting from scratch here (is this just for now or is this a long term choice you have made).

Reply:
ah you aren't passing via an args list: https://github.com/parksw3/epidist-paper/blob/d34a461c4d7526438d509eec99025abda30687c7/R/plot.R#L68

do the parameters you are passing influence what is being plot?

Reply:
Plotting from scratch because it's easier for me to write, rather than it being a long term thought out choice or something like that.

And yes that fixed it, great spot, thanks!